spellbook/models/x2y2/ethereum/x2y2_ethereum_events.sql

```diff
@@ -0,0 +1,180 @@
+{{ config(
+        alias ='events',
+        materialized ='incremental',
+        file_format ='delta',
+        incremental_strategy='merge',
+        unique_key='unique_trade_id'
+        )
+}}
+
+WITH aggregator_routed_x2y2_txs AS (
+    SELECT inv.evt_block_time AS block_time
+    , prof.to AS seller
+    , ROUND(bytea2numeric_v2(substring(get_json_object(inv.item, '$.data'), 195,64)),0) AS token_id
+    , get_json_object(inv.item, '$.price') AS amount_raw
+    , prof.currency AS currency_contract
+    , '0x' || substring(get_json_object(inv.item, '$.data'), 155, 40) AS project_contract_address
+    , tokens.name AS collection
+    , agg.name AS aggregator_name
+    , agg.contract_address AS aggregator_address
+    , inv.evt_tx_hash AS tx_hash
+    , prof.evt_index
+    , COALESCE(get_json_object(inv.item, '$.price')*get_json_object(get_json_object(inv.detail, '$.fees[0]'), '$.percentage')/1e6, 0) AS platform_fee_amount_raw
+    , COALESCE(get_json_object(get_json_object(inv.detail, '$.fees[0]'), '$.percentage')/1e6, 0) AS platform_fee_percentage
+    , COALESCE(get_json_object(inv.item, '$.price')*get_json_object(get_json_object(inv.detail, '$.fees[1]'), '$.percentage')/1e6, 0) AS royalty_fee_amount_raw
+    , COALESCE(get_json_object(get_json_object(inv.detail, '$.fees[1]'), '$.percentage')/1e6, 0) AS royalty_fee_percentage
+    , get_json_object(get_json_object(inv.detail, '$.fees[1]'), '$.to') AS royalty_fee_receive_address
+    FROM {{ source('x2y2_ethereum','X2Y2_r1_evt_EvProfit') }} prof
+    INNER JOIN {{ source('x2y2_ethereum','X2Y2_r1_evt_EvInventory') }} inv ON inv.itemHash = prof.itemHash
+    LEFT JOIN {{ ref('tokens_ethereum_nft') }} tokens ON ('0x' || substring(get_json_object(inv.item, '$.data'), 155, 40)) = tokens.contract_address
+    LEFT JOIN {{ ref('nft_ethereum_aggregators') }} agg ON agg.contract_address=taker
+    WHERE taker IN (SELECT contract_address FROM {{ ref('nft_ethereum_aggregators') }})
+    )
```

Comment:
add incremental filter here: 
{% if is_incremental() %} -- this filter will only be applied on an incremental run and prof.evt_block_time >= (select max(block_time) from {{ this }}) {% endif %}